pkg/ssh/ssh_internal_test.go

```diff
@@ -0,0 +1,44 @@
+package ssh
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"pgregory.net/rapid"
+)
+
+// Generates a random string that may contain the "Exit status <code>" string as a substring.
+func generateExitStatusString(t *rapid.T) (bool, string) {
+	prefix := rapid.String().Draw(t, "prefix")
+
+	exitStatus := ""
+	includeExitStatus := rapid.Bool().Draw(t, "includeExitStatus")
+	if includeExitStatus {
+		exitStatus = fmt.Sprintf("Exit status %s", ConnectionLimitReachedCode)
+	}
+
+	suffix := rapid.String().Draw(t, "suffix")
+
+	return includeExitStatus, fmt.Sprintf("%s%s%s", prefix, exitStatus, suffix)
+}
+
+func TestExitStatusWatcher(t *testing.T) {
+	t.Parallel()
+
+	rapid.Check(t, func(t *rapid.T) {
```

Comment:
Found a potential bug in the regex doing this. Modified the regex to accept only the number we care about (10001)